### PR TITLE
Generate node id file if not exists

### PR DESCRIPTION
**Pull request by mlohvynenko**

### CODE REVIEW

File: Kconfig [hunk 1]
```diff
@@ -47,6 +47,10 @@ config AOS_NODE_STATUS_FILE
 	string "Path to provisioning state file."
 	default "/lfs/aos/.nodestatus"
 
+config AOS_NODE_ID_FILE
+	string "Path to node id file."
+	default "/etc/machine-id"
```
Comment (al1img) [inline Kconfig:52]: default value should be /lfs/aos/node-id

File: boards/rpi_5.conf [hunk 2]
```diff
@@ -29,5 +29,6 @@ CONFIG_AOS_SERVICES_DIR="/1:/aos/services"
 CONFIG_AOS_NODE_CONFIG_FILE="/1:/aos/node_config.cfg"
 CONFIG_AOS_DISK_MOUNT_POINT="/1:"
 CONFIG_AOS_NODE_STATUS_FILE="/1:/aos/.nodestatus"
+CONFIG_AOS_NODE_ID_FILE="/1:/aos/.nodeid"
```
Comment (al1img) [inline boards/rpi_5.conf:32]: `/1:/aos/node-id"

File: src/nodeinfoprovider/nodeinfoprovider.cpp [hunk 3]
```diff
@@ -256,4 +239,32 @@ Error NodeInfoProvider::NotifyNodeStatusChanged(const NodeStatus& status)
     return ErrorEnum::eNone;
 }
 
+Error NodeInfoProvider::HandleFileNodeID()
```
Comment (al1img) [inline src/nodeinfoprovider/nodeinfoprovider.cpp:242]: No need for new function: put everything inside `InitNodeID`

File: src/nodeinfoprovider/nodeinfoprovider.cpp [hunk 4]
```diff
@@ -256,4 +239,32 @@ Error NodeInfoProvider::NotifyNodeStatusChanged(const NodeStatus& status)
...
+{
+    LOG_DBG() << "Read node id: path=" << cNodeIDFile;
+
+    auto err = FS::ReadFileToString(cNodeIDFile, mNodeInfo.mNodeID);
```
Comment (al1img) [inline src/nodeinfoprovider/nodeinfoprovider.cpp:246]: Handle error case first

File: src/nodeinfoprovider/nodeinfoprovider.cpp [hunk 5]
```diff
@@ -256,4 +239,32 @@ Error NodeInfoProvider::NotifyNodeStatusChanged(const NodeStatus& status)
...
+    mNodeInfo.mNodeID.Trim("\r\n");
+
+    if (err.IsNone() && !mNodeInfo.mNodeID.IsEmpty()) {
+        return ErrorEnum::eNone;
+    }
+
+    LOG_DBG() << "Generate node id";
+
+    auto uuid    = uuid::CreateUUID();
+    auto uuidStr = uuid::UUIDToString(uuid);
```
Comment (al1img) [inline src/nodeinfoprovider/nodeinfoprovider.cpp:257]: uuid::UUIDToString(uuid::CreateUUID());?

File: src/nodeinfoprovider/nodeinfoprovider.cpp [hunk 6]
```diff
@@ -256,4 +239,32 @@ Error NodeInfoProvider::NotifyNodeStatusChanged(const NodeStatus& status)
...
+
+    if (err = FS::WriteStringToFile(cNodeIDFile, uuidStr, S_IRUSR | S_IWUSR); !err.IsNone()) {
+        return AOS_ERROR_WRAP(err);
+    }
+
+    mNodeInfo.mNodeID = uuidStr;
+
+    LOG_DBG() << "Store node id: id=" << uuidStr << ", path=" << cNodeIDFile;
```
Comment (al1img) [inline src/nodeinfoprovider/nodeinfoprovider.cpp:265]: Would be nice to display node id in read case also

File: tests/nodeinfoprovider/src/main.cpp [hunk 7]
```diff
@@ -132,6 +139,29 @@ ZTEST(nodeinfoprovider, test_node_info)
     zassert_equal(nodeInfo.mAttrs[1].mValue, "xrun", "Attribute value mismatch");
 }
 
+ZTEST(nodeinfoprovider, test_node_id_is_generated)
+{
+    // remove node id file
```
Comment (al1img) [inline tests/nodeinfoprovider/src/main.cpp:144]: Remove this comment. The rule is: "Good code does not need comments".

File: tests/nodeinfoprovider/src/main.cpp [hunk 8]
```diff
@@ -132,6 +139,29 @@ ZTEST(nodeinfoprovider, test_node_info)
...
+    auto err = aos::FS::Remove(cNodeIDFile);
+    zassert_true(err.IsNone(), "Failed to remove node id file: %s", utils::ErrorToCStr(err));
+
+    aos::zephyr::nodeinfoprovider::NodeInfoProvider provider;
+
+    err = provider.Init();
+    zassert_true(err.IsNone(), "Failed to init node info provider: %s", utils::ErrorToCStr(err));
+
+    aos::StaticString<aos::cNodeIDLen> nodeID;
```
Comment (al1img) [inline tests/nodeinfoprovider/src/main.cpp:153]: Empty line after var definition.

File: tests/nodeinfoprovider/src/main.cpp [hunk 9]
```diff
@@ -132,6 +139,29 @@ ZTEST(nodeinfoprovider, test_node_info)
...
+    err = aos::FS::ReadFileToString(cNodeIDFile, nodeID);
+    zassert_true(err.IsNone(), "Failed to read node id from file: %s", utils::ErrorToCStr(err));
+
+    aos::NodeInfo nodeInfo;
```
Comment (al1img) [inline tests/nodeinfoprovider/src/main.cpp:157]: ditto

File: boards/rpi_5.conf (hunk 2 above)
Comment (al1img) [inline boards/rpi_5.conf:32]: Also set for native and native 64 boards.

**Review by al1img — APPROVED** [review]

**Review by mykola-kobets-epam — APPROVED** [review]

**Pull request merged**